librz/asm/p/asm_ppc_gnu.c

```diff
@@ -51,6 +52,14 @@ static int disassemble(RzAsm *a, RzAsmOp *op, const ut8 *buf, int len) {
 	/* prepare disassembler */
 	memset(&disasm_obj, '\0', sizeof(struct disassemble_info));
 	disasm_obj.disassembler_options = (a->bits == 64) ? "64" : "";
+	*options = 0;
+	if (!RZ_STR_ISEMPTY(a->cpu)) {
+		snprintf(options, sizeof(options), "%s,%s",
+			(a->bits == 64) ? "64" : "", a->cpu);
+	} else if (a->bits == 64) {
+		rz_str_ncpy(options, "64", sizeof(options));
+	}
+	disasm_obj.disassembler_options = options;
```

Comment:
If I'm not missing anything, `disasm_obj.disassembler_options` seems to be a `char *`, so assigning it to `options` which is a local variable is not ok.

Reply:
you need to alloc it

Reply:
`disasm_obj.disassembler_options = rz_str_newf("%s", options);`

Like that?

Reply:
Yes.